festim/hydrogen_transport_problem.py

```diff
@@ -579,12 +579,30 @@ def create_formulation(self):
                 * self.dx(source.volume.id)
             )
 
-            # add fluxes
-            # TODO implement this
-            # for bc in self.boundary_conditions:
-            #     pass
-            #     if bc.species == spe and bc.type != "dirichlet":
-            #         formulation += bc * v * self.ds
+        # add fluxes
+        # TODO implement this
+        # for bc in self.boundary_conditions:
+        #     pass
+        #     if bc.species == spe and bc.type != "dirichlet":
+        #         formulation += bc * v * self.ds
+
+        # check if each species is defined in all volumes
+        if not self.settings.transient:
+            for spe in self.species:
+                # if species mobile, already defined in diffusion term
+                if not spe.mobile:
+                    not_defined_in_volume = self.volume_subdomains.copy()
+                    for vol in self.volume_subdomains:
+                        # check reactions
+                        for reaction in self.reactions:
+                            if vol == reaction.volume:
+                                not_defined_in_volume.remove(vol)
+
+                    # if species not defined in all volumes, add to formulation
```

Comment:
add what to formulation?
Can you be more precise here for our future selves
```suggestion
                    # add c = 0 to formulation where needed
```